### ISSUE TITLE
Which frame do we want to calculate the Backstream and NonBackstream Pressures?

### ISSUE THREAD

**Issue by markusbattarbee** [fmihpc/vlasiator#382]
Currently we calculate the backstreaming and non-backstreaming pressure tensors by using the bulk V for that specific sub-population. @ursg argued that from the point of view of e.g. beam instabilities, this could give information on beam width. However, I argue that in global Vlasiator runs, in the quasi-parallel foreshock, the beam population evolution is so strong that the associated sub-population bulk V is going to be all over the place, and subsequently the pressure will be of little use. Thus, I suggest we change this to use the per-ion-population rest frame for calculating the pressure.

**Comment by ursg** [comment]
The quantity we are mostly looking at here is not the pressure itself, but the Temperature that's derived from it.

In the case of "backstreaming temperature", it's highly questionable to actually call it a temperature anyway, since the beam populations are, by definition, a nonthermal feature. In those cases, therefore, it's more of a guesswork-quantity to somewhat give us an idea of the beam extent. Not sure how useful it is now, and I aggree it might be of more benefit in terms of pressure wrt. the background.

The non-backstreaming temperature however continues to be useable as a kinda-proper temperature value, since it quantifies the thermal width of the (presumably roughly termal) core population. If you think about the quantities you, for example, feed to WHAMP to compare with analytical dispersion relations, this is exactly one of the input parameters, and it makes sense to have this calculated as the central second moment of the core population.

In other words: both have merit. Waaah. :)

**Comment by ykempf** [comment]
Thinking of a beam population with a high drift with respect to the core population, to me it doesn't make sense to compute a pressure in the core/bulk frame as this would yield a huge pressure. So indeed, arguments either way...

**Comment by ykempf** [comment]
http://ftp.issibern.ch/forads/sr-001-06.pdf Eq. 6.4 and 6.7

**Comment by ykempf** [comment]
If this is to be believed (and I have a feeling that I/we calculated some of that by hand one day) then we should be able to convert/translate/rotate to the frame we wish to use a posteriori, provided we store the relevant velocity vectors.
